upstream.wrap

```diff
@@ -1,6 +1,6 @@
 [wrap-file]
 directory = eigen-eigen-323c052e1731
```

Comment:
```
upstream.wrap directory 'eigen-eigen-323c052e1731' exists: ☒
  available file: eigen-3.3.7
```

Please try changing the directory to `eigen-3.3.7`

Reply:
locally this results in `../meson.build:11:0: ERROR: Subproject exists but has no meson.build file` but pushed it

Reply:
Interesting, I can't reproduce this. Do you have an example of a top level meson.build to reproduce?

```
$ cat meson.build 
project('myproj')

subproject('eigen-3.3.7')

$ ls subprojects/
eigen-3.3.7

$ ls subprojects/eigen-3.3.7/
bench           COPYING.BSD      COPYING.MPL2          debug  eigen3.pc.in  LICENSE.build  signature_of_eigen3_matrix_library
blas            COPYING.GPL      COPYING.README        demos  failtest      meson.build    test
cmake           COPYING.LGPL     CTestConfig.cmake     doc    INSTALL       README.md      unsupported
CMakeLists.txt  COPYING.MINPACK  CTestCustom.cmake.in  Eigen  lapack        scripts

$ meson builddir
The Meson build system
Version: 0.55.999
Source dir: /tmp/tmp.ugsy4ISP6C
Build dir: /tmp/tmp.ugsy4ISP6C/builddir
Build type: native build
Project name: myproj
Project version: undefined
Host machine cpu family: x86_64
Host machine cpu: x86_64

|Executing subproject eigen-3.3.7 method meson 
|
|Project name: eigen
|Project version: 3.3.7
|C++ compiler for the host machine: c++ (gcc 10.1.0 "c++ (GCC) 10.1.0")
|C++ linker for the host machine: c++ ld.bfd 2.34.0
|Build targets in project: 0
|Subproject eigen-3.3.7 finished.

Build targets in project: 0

myproj undefined

  Subprojects
    eigen-3.3.7: YES

Found ninja-1.10.0 at /usr/bin/ninja

```

Reply:
Is it possible that my local eigen.wrap file currently contains other contents than yours cause the patch_url, patch_filename, patch_hash are still from the old version for me locally so maybe we can just merge this and then it should work for me as well? I squashed and pushed. (cause I'm currently editing upstream.wrap and I'm not sure how my local eigen.wrap file provided by wrapdb is produced from the upstream.wrap here.

The source_filename looks a bit weird to me, so I changed that as well, hope this still works on your side?

Reply:
Is there a way for me to produce the same file as wrapdb serves, specifically generating and testing the patch.zip stuff locally, I'm wondering how that works/ gets produced. 
I also think this could include git patches (just thinking about meson wrap generally) or something like that but unsure whats actually happening behind the scenes, maybe I'm missing some docs or this could be added somewhere? (I mean technically I guess I could find the code somewhere :D)